src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    Random random = new Random();
+    ColorSupplier colorSupplier = new ColorSupplier();
+    Square square = new Square();
+    Circle circle = new Circle();
+    Rectangle rectangle = new Rectangle();
+    Triangle triangle = new Triangle();
+    Trapezoid trapezoid = new Trapezoid();
+    Figure[] figures = {square, circle, rectangle, triangle, trapezoid};
+
+    public Figure getRandomFigure() {
+        square.setColor(colorSupplier.getRandomColor());
+        square.setSide(random.nextInt(15)==0?5:random.nextInt(15));
+        circle.setColor(colorSupplier.getRandomColor());
+        circle.setRadius(random.nextInt(7)==0?7:random.nextInt(7));
+        rectangle.setColor(colorSupplier.getRandomColor());
+        rectangle.setHeight(random.nextInt(10)==0?5:random.nextInt(10));
+        rectangle.setWidth(random.nextInt(25)==0?10:random.nextInt(25));
+        triangle.setColor(colorSupplier.getRandomColor());
+        triangle.setBase(random.nextInt(10)==0?5:random.nextInt(10));
+        triangle.setHeight(random.nextInt(25)==0?5:random.nextInt(25));
+        trapezoid.setColor(colorSupplier.getRandomColor());
+        trapezoid.setTop(random.nextInt(7)==0?3:random.nextInt(7));
+        trapezoid.setBase(random.nextInt(20)==0?5:random.nextInt(20));
+        trapezoid.setHeight(random.nextInt(25)==0?10:random.nextInt(25));
+        int index = random.nextInt(figures.length);
+        return figures[index];
```

Comment:
Давай сделаем реалиацию через switch-case, тебе осталось только добавить приватные методы getCircle(), getIsoscelesTrapezoid(),.. которые будут возвращать соответствующие фигуры с случайными параметрами(сгенерированными при помощи Random)
```suggestion
        switch (random.nextInt(FIGURE_COUNT)) {
            case 0:
                return getCircle();
            case 1:
                return getIsoscelesTrapezoid();
            case 2:
                return getRectangle();
            case 3:
                return getRightTriangle();
            default:
                return getSquare();
        }
```

где FIGURE_COUNT - константа с колличеством фигур(private static final int FIGURE_COUNT = 5;)